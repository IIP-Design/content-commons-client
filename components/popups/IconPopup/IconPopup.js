/**
 *
 * IconPopup
 *
 */

import React from 'react';
import { string } from 'prop-types';
import { Icon, Popup } from 'semantic-ui-react';

import Focusable from 'components/Focusable/Focusable';

const IconPopup = props => {
  const {
    message, iconSize, iconType, popupSize,
  } = props;

  return (
    <Popup
      trigger={ (
        <span>
          <Focusable>
            <Icon size={ iconSize } name={ iconType } />
          </Focusable>
        </span>
      ) }
      content={ message }
      size={ popupSize }
      on={ [
        'hover',
        'click',
        'focus',
      ] }
      inverted
    />
  );
};

IconPopup.propTypes = {
  message: string.isRequired,
  iconSize: string,
  iconType: string.isRequired,
  popupSize: string,
};

export default IconPopup;
