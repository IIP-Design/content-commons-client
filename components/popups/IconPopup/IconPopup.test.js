import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import IconPopup from './IconPopup';

const props = {
  message: 'A test message',
  iconType: 'info circle',
  iconSize: 'tiny',
  popupSize: 'mini',
};

const newProps = {
  message: 'An updated test message',
  iconType: 'undo',
  iconSize: 'huge',
  popupSize: 'large',
};

const Component = <IconPopup { ...props } />;

describe( '<IconPopup />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'receives props', () => {
    const wrapper = shallow( Component );
    const { message, popupSize } = props;

    expect( wrapper.prop( 'content' ) ).toEqual( message );
    expect( wrapper.prop( 'size' ) ).toEqual( popupSize );
  } );

  it( 'sets new props', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { ...newProps } );
    expect( wrapper.prop( 'content' ) )
      .toEqual( newProps.message );
    expect( wrapper.prop( 'size' ) )
      .toEqual( newProps.popupSize );
  } );
} );
