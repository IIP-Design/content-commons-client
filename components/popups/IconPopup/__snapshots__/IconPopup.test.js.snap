// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconPopup /> renders without crashing 1`] = `
<Popup
  content="A test message"
  disabled={false}
  eventsEnabled={true}
  inverted={true}
  on={
    Array [
      "hover",
      "click",
      "focus",
    ]
  }
  pinned={false}
  popperModifiers={Array []}
  position="top left"
  size="mini"
  trigger={
    <span>
      <Focusable
        order={0}
      >
        <Icon
          as="i"
          name="info circle"
          size="tiny"
        />
      </Focusable>
    </span>
  }
/>
`;
